packages/react-instantsearch-core/src/core/createInstantSearchManager.js

```diff
@@ -342,6 +342,8 @@ export default function createInstantSearchManager({
     // to populate it on a custom key and override the `search` method to
     // search on it first.
     if (client.transporter) {
+      client._cacheHydrated = true;
```

Comment:
let's maybe add a test where we serialise twice?

Reply:
done